Sources/Frisbee/Interactors/Cancellable.swift

```diff
@@ -0,0 +1,7 @@
+public protocol Cancellable {
+    func cancel()
+}
+
+final class NilCancellable: Cancellable {
```

Comment:
What you think about `Uncancellable` for this class name?

Reply:
I was thinking and maybe this class is better to not exist. And instead `@discardableResults` in `get` functions it is better to use polimorphism overloading `get` function adding functions with `Cancelable` return.

Reply:
@ronanrodrigo I am not sure we can use polymorphism in this case. I used the NilCancellable in cases of `onComplete(.fail`. The function returns a task when success, but in case of failure we just return a class that will do nothin on `cancel` call. 

The name I chose was to recall the pattern. It should be NullCancellable, actually, but since it's Swift. I think Uncancellable will bring ubiquity to its behavior.

Reply:
Yes I understand your motivation and because of the `@discardableResult` who will use the Frisbee will not be affected by this decision. But an empty implementation of `cancel` method seems a code smell.

About the polymorphism, it is possible with we do that:
```swift
public final class NetworkGet: Getable {
    // ...
    public func get<T: Decodable, U: Encodable>(url: String, query: U, onComplete: @escaping OnComplete<T>) {
        guard let url = URL(string: url) else { return onComplete(.fail(.invalidUrl)) }
        return get(url: url, query: query, onComplete: onComplete) // 👈 it is necessary to use the "return" to avoid Ambiguous error
    }

    public func get<T: Decodable, U: Encodable>(url: URL, query: U, onComplete: @escaping OnComplete<T>) {
        // ...
        makeRequest(url: url, onComplete: onComplete)
    }

    public func get<T: Decodable, U: Encodable>(url: String, query: U, onComplete: @escaping OnComplete<T>) -> Cancelable {
        return makeRequest(url: url, onComplete: onComplete)
    }
}
```